mesonbuild/dependencies/dev.py

```diff
@@ -208,16 +208,23 @@ class LLVMDependencyConfigTool(ConfigToolDependency):
     __cpp_blacklist = {'-DNDEBUG'}
 
     def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
-        self.tools = get_llvm_tool_names('llvm-config')
-
-        # Fedora starting with Fedora 30 adds a suffix of the number
-        # of bits in the isa that llvm targets, for example, on x86_64
-        # and aarch64 the name will be llvm-config-64, on x86 and arm
-        # it will be llvm-config-32.
-        if environment.machines[self.get_for_machine_from_kwargs(kwargs)].is_64_bit:
-            self.tools.append('llvm-config-64')
+        for_machine = self.get_for_machine_from_kwargs(kwargs)
+
+        # Use config tool specified by LLVM_CONFIG, if defined
+        llvm_config_path = environment.lookup_binary_entry(for_machine, 'llvm_config')
+        if llvm_config_path:
+            self.tools = llvm_config_path
         else:
-            self.tools.append('llvm-config-32')
+            self.tools = get_llvm_tool_names('llvm-config')
```

Comment:
I don't think this is necessary, the first llvm_tool_name is already `'llvm-config'` and that's the entry that is looked up in a native file (and also what would be looked up via envconfig mappings).

If you remove this diff hunk and only modify `envconfig.py`, does it still work?

(By the way, builddir/meson-logs/meson-log.txt should have debug messages indicating how tool lookup was performed. `mesonbuild.programs.find_external_program` will be invoked with `name=self.tool_name, display_name=self.tool_name, default_names=self.tools`)

Reply:
You are right. I removed everything other than the `envconfig.py` changes (with the right binary name). It seems to work without any changes in `dev.py`